### ISSUE TITLE
Cookie bug: Multiple cookies are treated as 1 cookie

### ISSUE THREAD

**Issue by dannychu92** [Azure/azure-functions-dotnet-worker#1879]
When sending a request with cookies, then multiple cookies are treated as 1 cookie.
For example, if i send a http request, then in my f12 dev tools, i can see the "Cookie" header that is sent out.
The value of the "Cookie" header is: "foo=bar; hello=world".

Then in my api endpoint, HttpRequestData.Cookies, there are only 1 HttpCookie in the collection.
The HttpCookie.Name is "foo" and the HttpCookie.Value is "bar,hello=world".

I use Azure Functions .NET 7 isolated.
Nuget packages:

<PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.19.0" />
<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.1.0" />
<PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.14.1" />

**Comment by kshyju** [comment]
Sorry! I could not repro what you have described. To repro what you mentioned, I created an http trigger which reads the cookies and prints out. I can see the correct cookie values were retrieved.
Here is a POSTMAN screenshot from my testing
![image](https://github.com/Azure/azure-functions-dotnet-worker/assets/144469/56e0cd42-eb9c-434a-a747-480f0f80a8e3)

The function code I used is below. FWIW, I used the latest stable version of the Microsoft.Azure.Functions.Worker nuget packages.

```csharp
[Function("Function1")]
public HttpResponseData Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestData req)
{
    var response = req.CreateResponse(HttpStatusCode.OK);
    response.Headers.Add("Content-Type", "text/plain; charset=utf-8");

    var cookieCount = req.Cookies.Count;
    var cookies = req.Cookies.Select(c => $"cookie name:{c.Name}   cookie value:{c.Value}");

    var cookieStr = new StringBuilder();
    cookieStr.AppendLine("");
    foreach ( var cookie in cookies )
    {
        cookieStr.AppendLine(cookie);
    }

    response.WriteString($"Total {cookieCount}{cookieStr}");

    return response;
}
```

If you are still running into issues, Can you post the raw request header you used?

**Comment by dannychu92** [comment]
Hi @kshyju  , this is very puzzling.

When i tested this cookie thing, i tested through the browser with the normal "fetch()" api, and that made it seperate the cookies with a comma, so it became just 1 cookie.

But now i just tried with Postman, and the cookies were seperated with a semicolon so they work with Postman, but not in a browser (i tested on edge, firefox, chrome)

I did some more digging, and i think it has something to do with the "custom domain self-signed certificate" that i use.
For example, i created my own certificate for "myapp.me" and then added "127.0.0.1 myapp.me" in the "hosts" file.

Then i start my azure functions like this: 
func start --port 9800 --useHttps --cert "C:\ca\myapp.me\cert.pfx" --password "hello123" --build

And then if i go to "https://myapp.me:9800" and make an api call, then the cookies were bugged with the commma seperation.

If i start my azure function with a regular "func start", so my url is "http://localhost:7071", and do the same thing, then the cookie seperation works.

So maybe it has something to do with using a certificate?

I also just tried to test with a regular .net 7 asp.net core web app, which worked.
In the program.cs, i added this builder.WebHost.ConfigureKrestrel() to add my certificate:

```
builder.Services.AddRazorPages();
builder.WebHost.ConfigureKestrel(serverOptions =>
{
	serverOptions.Listen(IPAddress.Loopback, 5001, listenOptions =>
	{
		listenOptions.UseHttps(@"C:\ca\myapp.me\cert.pfx", "hello123");
	});
});
var app = builder.Build();
...
```

And then i create an api endpoint "/api/test"

And then i call GET "https://myapp.me:5001/api/test"
And then the cookies were seperated and works like expected.
So it works in Asp.net core with my own certificate through the browser, but not in Azure Functions through the browser.

So in summary:

Cookie seperation bugs when:
- I use my own certificate and make api request through browser, eg: GET https://myapp.me:9800/api/test

Cookie seperation works when:
- I use my own certificate and make api request through Postman, eg: GET https://myapp.me:9800/api/test
- I use the regular localhost "func start", and make api request through browser eg: GET http://localhost:7071/api/test

Note, i also tested on the latest nuget packages "Microsoft.Azure.Functions.Worker, version 1.19.0", and "Microsoft.Azure.Functions.Worker.Extensions.Http, version 3.1.0"

**Comment by kshyju** [comment]
Can you capture a fiddler trace when it fails from the fetch request and inspect that request for the headers? Do they look correct you?

**Comment by pavel-dushkov-am** [comment]
I also faced this problem.
The initial conditions are almost the same:
- "Microsoft.Azure.Functions.Worker, version 1.19.0", and "Microsoft.Azure.Functions.Worker.Extensions.Http, version 3.1.0"
- created my own certificate for "myapp.me"
- added "127.0.0.1 myapp.me" in the "hosts" file
- started my azure functions like `func start --port 9800 --useHttps --cert "C:\ca\myapp.me\cert.pfx" --password "hello123"`

and when I'm making request through browser, azure function receives cookies that separated by coma, even though I can see in the developer tools in Request Headers section, that the sent cookies are separated by a semicolon.
If I copy this request as CURL and make it through Postman, then azure function receives cookies that separated by semicolon, as I sent it.

I tried to catch the traffic in the fiddler to find the source of the problem. But unfortunately, when traffic capture is enabled in the fiddler, the problem does not reproduce - even from the browser, cookies begin to arrive in the correct form (semicolon separated).
I came across an issue with a similar problem, where fiddler had the same effect - https://github.com/microsoft/reverse-proxy/issues/437
